src/lib/problem_report.c

```diff
@@ -305,13 +305,16 @@ load_stream(FILE *fp)
 
 #define MAX_OPT_DEPTH 10
 static int
-format_percented_string(const char *str, problem_data_t *pd, FILE *result)
+format_percented_string(const char *str, problem_data_t *pd, FILE *result, char *fmt_file)
 {
     long old_pos[MAX_OPT_DEPTH] = { 0 };
     int okay[MAX_OPT_DEPTH] = { 1 };
     long len = 0;
     int opt_depth = 1;
 
+    char *missing_item = xstrdup(str);
```

Comment:
I feel like this is unnecessary. What about null-initializing it here and duplicating lazily as errors are detected?